awx_collection/tests/integration/targets/group/tasks/main.yml

```diff
@@ -117,9 +122,10 @@
     state: absent
   register: result
 
+# In this case, group 2 was last a child of group1 so deleting group1 deleted group2
 - assert:
     that:
-      - "result is changed"
+      - "result is not changed"
```

Comment:
Confirming - this was a preexisting failure.

Reply:
I think it used to work at one point but if you try this scenario in the UI now you get two options:
* delete the child group(s) (default)
* promote the child group(s)

The code in the group module follows the first bullet now. So when we delete group1 first group2 is now gone.